src/communication/channelmanager.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <aos/common/tools/timer.hpp>
+
+#include "channel.hpp"
+#include "channelmanager.hpp"
+#include "log.hpp"
+
+namespace aos::zephyr::communication {
+
+Error ChannelManager::Init(TransportItf& transport)
+{
+    LOG_INF() << "Init channel manager";
+
+    mTransport = &transport;
+
+    return Run();
+}
+
+RetWithError<ChannelItf*> ChannelManager::CreateChannel(uint32_t port)
+{
+    UniqueLock lock(mMutex);
+
+    LOG_DBG() << "Create channel port=" << port;
+
+    ChannelInfo info = {port, info.mChannel = aos::MakeShared<Channel>(&mChanAllocator, mTransport, port)};
```

Comment:
comparing CreateChannel &  DeleteChannel implementation. Manager should create a single channel per port. It means create function should check whether channel with given port exists & return existing one.

Reply:
Done